File: web/src/api/components/schemas/Episode.ts
import { Option } from 'fp-ts/lib/Option';
import { Int, string, boolean, array, type } from 'io-ts';
import { DateFromISOString } from 'io-ts-types/lib/DateFromISOString';
import { optionFromNullable } from 'io-ts-types/lib/optionFromNullable';

export type Episode = {
	id: Option<Int>;
	showId: Option<Int>;
	showName: Option<string>;
	showSlug: Option<string>;
	slug: Option<string>;
	epNo: Option<string>;
	title: Option<string>;
	description: Option<string>;
	publishedAt: Option<Date>;
	durationSec: Option<Int>;
	views: Option<string>;
	videoUrl: Option<string>;
	thumbMediaId: Option<Int>;
	imageUrl: Option<string>;
	live: Option<boolean>;
	tags: Option<Array<string>>;
	published: Option<boolean>;
	createdAt: Option<Date>;
	updatedAt: Option<Date>;
};
export const EpisodeIO = type(
	{
		id: optionFromNullable(Int),
		showId: optionFromNullable(Int),
		showName: optionFromNullable(string),
		showSlug: optionFromNullable(string),
		slug: optionFromNullable(string),
		epNo: optionFromNullable(string),
		title: optionFromNullable(string),
		description: optionFromNullable(string),
		publishedAt: optionFromNullable(DateFromISOString),
		durationSec: optionFromNullable(Int),
		views: optionFromNullable(string),
		videoUrl: optionFromNullable(string),
		thumbMediaId: optionFromNullable(Int),
		imageUrl: optionFromNullable(string),
		live: optionFromNullable(boolean),
		tags: optionFromNullable(array(string)),
		published: optionFromNullable(boolean),
		createdAt: optionFromNullable(DateFromISOString),
		updatedAt: optionFromNullable(DateFromISOString),
	},
	'Episode',
);

File: backend/src/main/java/net/tape/model/Episode.java
package net.tape.model;

import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;
import java.time.Instant;

/** Domain model — framework-free POJO. */
@Getter
@Setter
public class Episode implements Serializable {
    private Long id;
    private Long showId;
    private String showName;
    private String showSlug;
    private String slug;
    private String epNo;
    private String title;
    private String description;
    private Instant publishedAt;
    private Integer durationSec;
    private String views;
    private String videoUrl;
    private Long thumbMediaId;
    private String imageUrl;
    private boolean live;
    private String[] tags;
    private boolean published;
    private Instant createdAt;
    private Instant updatedAt;
}
